bitcoin/src/crypto/ecdsa.rs

```diff
@@ -202,10 +202,10 @@ impl fmt::Display for Error {
         use Error::*;
 
         match *self {
-            HexEncoding(ref e) => write_err!(f, "signature hex encoding error"; e),
```

Comment:
Heh, when did this macro get introduced. It doesn't print the inner error anymore in the error.. That's very non-standard and I really don't like that.. That means in a panic you won't see the underlying error anymore... The macro seems to assume the person dealing with the error has access to `.source()` which is almost never the case lol, that's why it's printed into the `Display`..

Reply:
I'll defer to @Kixunil to debate the finer points of error handling :)

Reply:
@stevenroose it's always the case for people using `anyhow`, which is the majority of actual applications that end-users see.

If we stick the outer error into our `Display` implementations then it's impossible for any downstream users to remove it so they're screwed if they want to provide a better error display with more context.

What we're doing has been standard practice for many years now. The only thing non-standard is that we re-introduce the prefix when `std` is disabled, since then we don't have `source`.